chain/client/src/view_client.rs

```diff
@@ -805,3 +805,27 @@ impl Handler<GetGasPrice> for ViewClientActor {
             .map_err(|e| e.to_string())
     }
 }
+
+/// Starts the View Client in a new arbiter (thread).
+pub fn start_view_client(
+    validator_account_id: Option<AccountId>,
+    chain_genesis: ChainGenesis,
+    runtime_adapter: Arc<dyn RuntimeAdapter>,
+    network_adapter: Arc<dyn NetworkAdapter>,
+    config: ClientConfig,
+) -> Addr<ViewClientActor> {
+    SyncArbiter::start(config.view_client_threads, move || {
```

Comment:
Just to be completely clear on this. We will be creating `X` ViewClientActor instances, running "independently" from each other. I'm concerned for example about this fields:

```
    /// Transaction query that needs to be forwarded to other shards
    pub tx_status_requests: SizedCache<CryptoHash, Instant>,
    /// Transaction status response
    pub tx_status_response: SizedCache<CryptoHash, FinalExecutionOutcomeView>,
    /// Query requests that need to be forwarded to other shards
    pub query_requests: SizedCache<String, Instant>,
    /// Query responses from other nodes (can be errors)
    pub query_responses: SizedCache<String, Result<QueryResponse, String>>,
    /// Receipt outcome requests
    pub receipt_outcome_requests: SizedCache<CryptoHash, Instant>,
```

since they are not shared between instances (which we don't want to do anyway or else all this threading is pointless) some messages can go to one instance first, and then to the other, in such a way that they make progress 1 / threads slower than before.

If this doesn't make any sense for ViewClient let me know and mark as resolved.

Reply:
Good point. Even though for single shard it doesn't matter, we should still put some thought into making it work properly @ilblackdragon

Reply:
Oh, you are right. The requests/responses should be in a single actor. Let me fix this.